src/ring_theory/dedekind_domain/adic_valuation.lean

```diff
@@ -333,4 +333,72 @@ instance adic_completion.has_lift_t : has_lift_t K (v.adic_completion K) :=
 /-- The ring of integers of `adic_completion`. -/
 def adic_completion_integers : valuation_subring (v.adic_completion K) := valued.v.valuation_subring
 
+variables (R K)
+
+lemma adic_completion.is_integer (x : v.adic_completion K) :
+  x ∈ v.adic_completion_integers K ↔ (valued.v x : ℤₘ₀) ≤ 1 :=
+iff.rfl
+
+/-- The natural inclusion of `R` in `adic_completion_integers`, as a ring homomorphism. -/
+@[simps] def inj_adic_completion_integers : R →+* v.adic_completion_integers K :=
```

Comment:
I don't think we need to name this map. Given that it is registered as part of the canonical `algebra` structure below, it will be accessible as `algebra_map R (v.adic_completion_integers K)`.